flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/SourceStreamTaskTest.java

```diff
@@ -479,5 +481,40 @@ public void cancel() {
 			running = false;
 		}
 	}
+
+	private static final class OutputRecordInCloseTestSource extends RichSourceFunction<String> {
```

Comment:
Make this source also implement `BoundedOneInput` to further verify that case?

Reply:
I agree.